python/ray/tests/accelerators/test_hpu.py

```diff
@@ -113,6 +116,216 @@ def test_check_accelerator_info():
+    ray.get(actor.set_habana_visible_devices.remote("0,1"))
+    assert ray.get(actor.get_habana_visible_devices.remote()) == "0,1"
+
+
+def test_hpu_ids(shutdown_only):
+    num_hpus = 3
+    ray.init(num_cpus=num_hpus, resources={"HPU": num_hpus})
+
+    def get_hpu_ids(hpus_per_worker):
+        hpu_ids = ray.get_runtime_context().get_accelerator_ids()["HPU"]
+        assert len(hpu_ids) == hpus_per_worker
+        modules = os.environ.get("HABANA_VISIBLE_MODULES")
+        if modules is not None:
+            assert modules == ",".join([str(i) for i in hpu_ids])  # noqa
+        for hpu_id in hpu_ids:
+            assert hpu_id in [str(i) for i in range(num_hpus)]
+        return hpu_ids
+
+    f0 = ray.remote(resources={"HPU": 0})(lambda: get_hpu_ids(0))
+    f1 = ray.remote(resources={"HPU": 1})(lambda: get_hpu_ids(1))
+
+    # Wait for all workers to start up.
+    @ray.remote
+    def g():
+        time.sleep(0.2)
+        return os.getpid()
+
+    start_time = time.time()
+    while True:
+        num_workers_started = len(set(ray.get([g.remote() for _ in range(num_hpus)])))
+        if num_workers_started == num_hpus:
+            break
+        if time.time() > start_time + 10:
+            raise RayTestTimeoutException(
+                "Timed out while waiting for workers to start up."
+            )
+
+    list_of_ids = ray.get([f0.remote() for _ in range(10)])
+    assert list_of_ids == 10 * [[]]
+    ray.get([f1.remote() for _ in range(10)])
+
+    # Test that actors have NEURON_RT_VISIBLE_CORES set properly.
+
+    @ray.remote
+    class Actor0:
+        def __init__(self):
+            hpu_ids = ray.get_runtime_context().get_accelerator_ids()["HPU"]
+            assert len(hpu_ids) == 0
+            assert os.environ["HABANA_VISIBLE_MODULES"] == ",".join(
+                [str(i) for i in hpu_ids]  # noqa
+            )
+            # Set self.x to make sure that we got here.
+            self.x = 0
+
+        def test(self):
+            hpu_ids = ray.get_runtime_context().get_accelerator_ids()["HPU"]
+            assert len(hpu_ids) == 0
+            assert os.environ["HABANA_VISIBLE_MODULES"] == ",".join(
+                [str(i) for i in hpu_ids]  # noqa
+            )
+            return self.x
+
+    @ray.remote(resources={"HPU": 1})
+    class Actor1:
+        def __init__(self):
+            hpu_ids = ray.get_runtime_context().get_accelerator_ids()["HPU"]
+            assert len(hpu_ids) == 1
+            assert os.environ["HABANA_VISIBLE_MODULES"] == ",".join(
+                [str(i) for i in hpu_ids]  # noqa
+            )
+            # Set self.x to make sure that we got here.
+            self.x = 1
+
+        def test(self):
+            hpu_ids = ray.get_runtime_context().get_accelerator_ids()["HPU"]
+            assert len(hpu_ids) == 1
+            assert os.environ["HABANA_VISIBLE_MODULES"] == ",".join(
+                [str(i) for i in hpu_ids]
+            )
+            return self.x
```

Comment:
This can be combined as a single Actor definition where number of hpus is input to the constructor.